ts/src/header-validator/validate-json.ts

```diff
@@ -561,7 +561,21 @@ function filterDataKeyValue(
     return None
   }
 
-  return set(ctx, j, string, {
+  const filterStringLength = (s: string) => {
```

Comment:
It may be easier to understand the errors here with an optional prefix so that the key being too long can be more easily distinguished from the value being too long, e.g.

```ts
const filterStringLength = (s: string, errPrefix: string = '') => {
  ...
  ctx.error(`${errPrefix}exceeds...`)

...

if (!filterStringLength(key, 'key ')) {
```

Same for the aggregation keys below.

Reply:
Is that necessary with the path already?

Reply:
Yes, because the path does not contain an element that distinguishes key from value.

Reply:
Updated with the error prefix, thanks!